tests/model_inheritance/models.py

```diff
@@ -186,3 +186,43 @@ class Child(Parent):
 
 class GrandChild(Child):
     pass
+
+
+class CommonAncestor(models.Model):
+    id = models.IntegerField(primary_key=True, default=1)
+
+
+class FirstParent(CommonAncestor):
+    first_ancestor = models.OneToOneField(
+        CommonAncestor, models.CASCADE, primary_key=True, parent_link=True
+    )
+
+
+class SecondParent(CommonAncestor):
+    second_ancestor = models.OneToOneField(
+        CommonAncestor, models.CASCADE, primary_key=True, parent_link=True
+    )
+
+
+class CommonChild(FirstParent, SecondParent):
+    pass
+
+
+class FirstParent2(models.Model):
+    first_parent_id = models.IntegerField(primary_key=True, default=1)
+
+
+class SecondParent2(models.Model):
+    second_parent_id = models.AutoField(primary_key=True)
+
+
+class CommonChild2(FirstParent2, SecondParent2):
+    pass
+
+
+class GrandCommonChild(CommonChild2):
+    pass
+
+
+class GrandCommonChildManyParents(GrandCommonChild, FirstParent2):
+    pass
```

Comment:
There is no need for all new models. It was enough to add `ItalianRestaurantManyParents(ItalianRestaurant, Place`.